exercises/tree-building/src/main/BuildTree.java

```diff
@@ -0,0 +1,72 @@
+import java.util.ArrayList;
```

Comment:
This should not contain the entire implementation. Please see some of the other exercises as an example and see the instructions in our [contributing guide](https://github.com/exercism/java/blob/master/CONTRIBUTING.md#adding-a-new-exercise) 🙂

Reply:
yes but i notice in the problem specification, this exercise is a bit special, it is about `Refactor` a tree building algorithm. Thus in general a naive solution would be given and the task is to refactor it into a more elegant and efficient solution.

Reply:
Ah, fair enough! I missed that, sorry 🙂